janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto

```diff
@@ -21,6 +21,49 @@ import "google/protobuf/wrappers.proto";
 option java_multiple_files = true;
 option java_package = "org.janusgraph.graphdb.grpc.types";
 
+enum PropertyDataType {
+    String = 0;
+    Character = 1;
+    Boolean = 2;
+    Int8 = 3;
+    Int16 = 4;
+    Int32 = 5;
+    Int64 = 6;
+    Float32 = 7;
+    Float64 = 8;
+    Date = 9;
+    JavaObject = 10;
+    GeoShape = 11;
+    Uuid = 12;
+}
+
+enum VertexPropertyCardinality {
+    Single = 0;
+    List = 1;
+    Set = 2;
+}
+
+enum PropertyBinding {
+    Globally = 0;
+    Local = 1;
```

Comment:
What does local mean?

Reply:
Bound to a vertex/edge.

Reply:
Got it, but this seems to contradict to the comments below:

> // * schema constraints: property is globally defined but bind to vertex or edge

From which I inferred whether a property is bound to a vertexLabel/edgeLabel has nothing to do with whether its global or loca.

Reply:
I've never used the binding feature myself before, so I might be missing something here. The "two more cases" you mentioned,

> allow different datatypes for different vertexLabel with the same property name.

and 

> Global properties (can be used on all vertexLabels) with activated schema constraint

are they legitimate cases in JanusGraph now? Also, I don't get the latter case. Can you elaborate or point me to doc/code?

Reply:
I thought about possible future ideas for Schema Management.

Reply:
I thought of possible future use cases how we can define properties to be more flexible.

Reply:
I tried to update the schema. 
@li-boxuan @rngcntr  what do you think about my updated version?

Reply:
To be honest I still don't understand the difference between

[this]( https://github.com/JanusGraph/janusgraph/pull/2561/files#diff-9c648aed04a9a8f61584cfb812ecc2727a893f6fd2a486502e75e8b3beab7f16R58-R59):
> Property is defined a side of a vertex or edge.

and 

[this:](https://github.com/JanusGraph/janusgraph/pull/2561/files#diff-9c648aed04a9a8f61584cfb812ecc2727a893f6fd2a486502e75e8b3beab7f16R63)
> schema constraints: property is globally defined but bind to vertex or edge

Reply:
I talked to @farodin91 about the intention of this feature earlier today. To wrap it up, the idea is that although a property key (say, `direction` for example) can have different constraints for different vertex labels. For instance, the `direction` property on a vertex labeled `speaker` could be constrained to values `left` and `right`. However, the same property `direction` on a vertex labeled `entrance` could be constrained to values `north`, `east`, `south` and `west`.
That would be enabled by a local binding.